segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/BaseChunkStorageProvider.java

```diff
@@ -0,0 +1,592 @@
+     *
+     * @return True or false.
+     */
+    @Override
+    abstract public boolean supportsNativeConcat();
+
+
+    /**
+     * Determines whether named file/object exists in underlying storage.
+     *
+     * @param chunkName Name of the storage object to check.
+     * @return True if the object exists, false otherwise.
+     */
+    @Override
+    final public boolean exists(String chunkName)  throws IOException {
+        // Validate parameters
+        Preconditions.checkArgument(null != chunkName, "chunkName must not be null");
+
+        long traceId = LoggerHelpers.traceEnter(log, "exists", chunkName);
+
+        // Call concrete implementation.
+        boolean retValue = doesExist(chunkName);
+
+        LoggerHelpers.traceLeave(log, "exists", traceId, chunkName);
+        return retValue;
+    }
+
+
+    /**
+     * Creates a new file.
+     *
+     * @param chunkName String name of the storage object to create.
+     * @return ChunkHandle A writable handle for the recently created chunk.
+     * @throws IOException Throws IOException in case of I/O related exceptions.
+     */
+    @Override
+    final public ChunkHandle create(String chunkName) throws IOException {
+        // Validate parameters
+        Preconditions.checkArgument(null != chunkName, "chunkName must not be null");
+
+        long traceId = LoggerHelpers.traceEnter(log, "create", chunkName);
+        Timer timer = new Timer();
+
+        // Call concrete implementation.
+        ChunkHandle handle = doCreate(chunkName);
+
+        // Record metrics.
+        Duration elapsed = timer.getElapsed();
+        CREATE_LATENCY.reportSuccessEvent(elapsed);
+        CREATE_COUNT.inc();
+
+        log.debug("Create chunk={} latency={}.", chunkName, elapsed.toMillis());
+        LoggerHelpers.traceLeave(log, "create", traceId, chunkName);
+        return handle;
+    }
+
+    /**
+     * Deletes a file.
+     *
+     * @param handle ChunkHandle of the storage object to delete.
+     * @return True if the object was deleted, false otherwise.
+     * @throws IOException Throws IOException in case of I/O related exceptions.
+     */
+    @Override
+    final public boolean delete(ChunkHandle handle) throws IOException {
+        // Validate parameters
+        Preconditions.checkArgument(null != handle, "handle must not be null");
+        Preconditions.checkArgument(!handle.isReadOnly(), "handle must not be readonly");
+        long traceId = LoggerHelpers.traceEnter(log, "delete", handle.getChunkName());
+        Timer timer = new Timer();
+
+        // Call concrete implementation.
+        boolean retValue = doDelete(handle);
+
+        // Record metrics.
+        Duration elapsed = timer.getElapsed();
+        DELETE_LATENCY.reportSuccessEvent(elapsed);
+        DELETE_COUNT.inc();
+
+        log.debug("Delete chunk={} latency={}.", handle.getChunkName(), elapsed.toMillis());
+        LoggerHelpers.traceLeave(log, "delete", traceId, handle.getChunkName());
```

Comment:
There's no point in doing trace enter-leave if you log-debug everything. I would suggest cleaning it up.

Reply:
`log.debug("Delete chunk={} latency={}.", handle.getChunkName(), elapsed.toMillis());` 
has latency information which trace does not include.
Recently, we found such information to be useful for debugging with ECS.